### PR TITLE
Misc quality improvements

### PR DESCRIPTION
**Pull request by ctz**
Some things from my server support branch that can be landed now.

### CODE REVIEW

File: rustls-libssl/admin/matrix.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
+
+# from openssl 3.0.0
```
Comment (cpu) [inline rustls-libssl/admin/matrix.py:3]: nit: maybe "from openssl 3.0.0" -> "from openssl 3.0.0 util/libssl.num"

Saves the next person having to find it in the src tree.

File: rustls-libssl/src/x509.rs [hunk 2]
```diff
@@ -40,7 +42,7 @@ impl OwnedX509Stack {
 impl Drop for OwnedX509Stack {
     fn drop(&mut self) {
         unsafe {
-            OPENSSL_sk_free(self.raw as *mut OPENSSL_STACK);
+            OPENSSL_sk_pop_free(self.raw as *mut OPENSSL_STACK, Some(X509_free));
```
[the remark this answers is not in the document]
Comment (ctz) [reply rustls-libssl/src/x509.rs:45]: Correct, yes.

File: rustls-libssl/src/lib.rs [hunk 3]
```diff
@@ -109,7 +109,7 @@ impl SslCipher {
     }
 
     pub fn protocol_id(&self) -> u16 {
-        self.rustls.suite().get_u16()
+        u16::from(self.rustls.suite())
```
[the remark this answers is not in the document]
Comment (ctz) [reply rustls-libssl/src/lib.rs:112]: This was actually an API change in rustls: https://github.com/rustls/rustls/commit/4f71a2b82e23ab86f493b0006dbe6c75224dbf46

[the remark this answers is not in the document]
Comment (cpu) [reply rustls-libssl/src/lib.rs:112]: Ohhhh right! Thanks, my mistake.

File: rustls-libssl/src/x509.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (cpu) [reply rustls-libssl/src/x509.rs:45]: Just curious: was this caught by valgrind or noticed manually?

File: rustls-libssl/admin/matrix.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        if next_line and 'pub fn ' in line:
+            parts = line.strip().replace('(', ' ').split()
+            items[parts[2][1:]] = not stub
+
+    return items
+
+impls = read_impls()
+
+# from ubuntu curl 7.81.0-1ubuntu1.15
+# extracted by running with LD_DEBUG=all LD_USE_LOAD_BIAS=0
```
[the remark this answers is not in the document]
Comment (ctz) [reply rustls-libssl/admin/matrix.py:59]: I'm afraid I'm going to claim the fifth on this one. There's some rationale on https://stackoverflow.com/questions/12479208/list-used-symbols-of-a-certain-dynamically-loaded-library but I don't claim to fully understand the implications.

[the remark this answers is not in the document]
Comment (ctz) [reply rustls-libssl/admin/matrix.py:59]: (It seems to make no difference with or without, so I'll remove this.)

[the remark this answers is not in the document]
Comment (cpu) [reply rustls-libssl/admin/matrix.py:59]: Interesting, seems specific to "prelinking" (TIL). 

Not important in any case, just being curious. Thanks for the SO link.

File: rustls-libssl/src/x509.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ctz) [reply rustls-libssl/src/x509.rs:45]: Manually.

Though this comment got me thinking -- what is the valgrind job currently doing? The answer is currently not much -- PR for that is incoming!

**Pull request merged**